MFVNeutralino/plugins/VertexTracksGen.cc

```diff
@@ -0,0 +1,401 @@
+#include "TH2.h"
+#include "CLHEP/Random/RandBinomial.h"
+#include "CLHEP/Random/RandomEngine.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "DataFormats/JetReco/interface/PFJetCollection.h"
+#include "DataFormats/JetReco/interface/GenJet.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/PatCandidates/interface/Jet.h"
+#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "FWCore/Framework/interface/EDFilter.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Utilities/interface/RandomNumberGenerator.h"
+#include "TrackingTools/IPTools/interface/IPTools.h"
+#include "TrackingTools/Records/interface/TransientTrackRecord.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrack.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrackBuilder.h"
+#include "JMTucker/Tools/interface/AnalysisEras.h"
+#include "JMTucker/Tools/interface/TrackRescaler.h"
+#include "JMTucker/MFVNeutralinoFormats/interface/MCInteractions.h"
+
+class MFVVertexTracksGen : public edm::EDFilter {
```

Comment:
Do I remember correctly that this was an attempt to match tracks to gen-level particles? I can't remember the outcome of this study (but I do remember Shaun and I having difficulty with this in the past).

Reply:
Yes, I made this producer to make those matched tracks as input to vertexer, so that helps us to use partially "gen" level tracks to fit SVs. But even doing that doesn't help me get 2 SVs for those signal samples. That is the reason why I didn't move on to save more seed tracks.